lib/puppet/module/task.rb

```diff
@@ -0,0 +1,85 @@
+require 'puppet/util/logging'
+require 'json'
+
+class Puppet::Module
+  class Task
+    class Error < Puppet::Error; end
+    class InvalidName < Error; end
+    class InvalidFile < Error; end
+    class TaskNotFound < Error; end
+
+    FORBIDDEN_EXTENSIONS = %w{.conf .md}
+
+    def self.is_task_name?(name)
```

Comment:
Naively, the name of this method implies to me that I'm checking whether puppet knows of a task by this name. What about renaming to something like `self.valid_task_name?(name)`